### PR TITLE
feat: add FluxRecord.Row with response data stored in List

### PR DESCRIPTION
**Pull request by michaelahojna**
Closes #

## Proposed Changes

Adding possibility of accessing response data in List FluxRecord.Row. 

In case of using pivot on data, where field contains labels that occur by default in the annotated CSV (f.e. "result" or "table"), could be duplicated column names in response. In that case FluxRecord.Values (Dictionary), which can hold only unique keys, doesn't show complete data. This edge case is solved by using FluxRecord.Row (List).

## Checklist

- [x] CHANGELOG.md updated
- [x] Rebased/mergeable
- [x] A test has been added if appropriate
- [x] `dotnet test` completes successfully
- [x] Commit messages are [conventional](https://www.conventionalcommits.org/en/v1.0.0/)
- [x] Sign [CLA](https://influxdata.com/community/cla/) (if not already signed)

### CODE REVIEW

File: Client.Core/Flux/Domain/FluxRecord.cs [hunk 1]
```diff
@@ -23,6 +23,23 @@ public class FluxRecord
         /// </summary>
         public Dictionary<string, object> Values { get; } = new Dictionary<string, object>();
 
+        /// <summary>
+        /// The record's rows.
+        /// </summary>
+        public List<Row> Rows { get; } = new List<Row>();
```
Comment (bednar) [inline Client.Core/Flux/Domain/FluxRecord.cs:29]: Please use simple `object` instead of `Row`:

```suggestion
        public List<object> Row { get; } = new List<object>();
```

File: Client.Core/Flux/Internal/FluxCsvParser.cs [hunk 2]
```diff
@@ -267,7 +267,14 @@ private FluxRecord ParseRecord(int tableIndex, FluxTable table, CsvReader csv)
...
                 var strValue = csv[fluxColumn.Index + 1];
 
+                if (record.Values.ContainsKey(columnName))
+                {
+                    record.Values.Remove(columnName);
+                }
+
                 record.Values.Add(columnName, ToValue(strValue, fluxColumn));
+
+                record.Rows.Add(new FluxRecord.Row(columnName, ToValue(strValue, fluxColumn)));
```
Comment (bednar) [inline Client.Core/Flux/Internal/FluxCsvParser.cs:277]: Please parse value only once ... call `ToValue(strValue, fluxColumn)` only once.

File: Client.Core/Flux/Internal/FluxCsvParser.cs [hunk 3]
```diff
@@ -389,6 +396,16 @@ private void AddColumnNamesAndTags(FluxTable table, CsvReader columnNames)
...
             }
+
+            var duplicates = table.Columns.GroupBy(col => col.Label)
+                .Where(rec => rec.Count() > 1)
+                .Select(label => label.Key).ToList();
+            if (duplicates.Any())
+            {
+                Console.WriteLine(
+                    $"The response contains columns with duplicated names: {string.Join(", ", duplicates)}\n" +
+                    "You should use the 'record.row' to access your data instead of 'record.values' dictionary.");
```
Comment (bednar) [inline Client.Core/Flux/Internal/FluxCsvParser.cs:407]: Please use Camel case for names of fields:

```suggestion
                    "You should use the 'record.Row' to access your data instead of 'record.Values' dictionary.");
```

File: Examples/RecordRowExample.cs [hunk 4]
```diff
@@ -0,0 +1,68 @@
+using System;
+using System.Linq;
+using System.Threading.Tasks;
+using InfluxDB.Client;
+using InfluxDB.Client.Api.Domain;
+
+
+namespace Examples
+{
+    public static class RecordRowExample
```
Comment (bednar) [inline Examples/RecordRowExample.cs:10]: Please add this example into `Examples/README.md` with proper description.

File: Client.Core/Flux/Internal/FluxCsvParser.cs [hunk 5]
```diff
@@ -267,7 +267,14 @@ private FluxRecord ParseRecord(int tableIndex, FluxTable table, CsvReader csv)
 
                 var strValue = csv[fluxColumn.Index + 1];
 
+                if (record.Values.ContainsKey(columnName))
+                {
+                    record.Values.Remove(columnName);
+                }
+
                 record.Values.Add(columnName, ToValue(strValue, fluxColumn));
```
Comment (bednar) [inline Client.Core/Flux/Internal/FluxCsvParser.cs:275]: We can simplify the code to: `record.Values[columnName] = ToValue(strValue, fluxColumn);`

**Review by bednar — CHANGES_REQUESTED** [review]
Thanks for your PR 👍 
There are a few requirements that must be be satisfy before we accept the PR:

File: Client.Core/Flux/Domain/FluxRecord.cs [hunk 6]
```diff
@@ -26,7 +26,7 @@ public class FluxRecord
         /// <summary>
         /// The record's rows.
         /// </summary>
-        public List<KeyValuePair<string, object>> Rows { get; } = new List<KeyValuePair<string, object>>();
+        public List<KeyValuePair<string, object>> Row { get; } = new List<KeyValuePair<string, object>>();
```
Comment (bednar) [inline Client.Core/Flux/Domain/FluxRecord.cs:29]: Please use `List<object>` without column names. The row should contains pure response from InfluxDB without additional data.

```suggestion
        public List<object> Row { get; } = new List<object>();
```

**Review by bednar — CHANGES_REQUESTED** [review]

File: Examples/RecordRowExample.cs [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+            const string password = "my-password";
+            const string bucket = "my-bucket";
+            const string org = "my-org";
+
+            using var client = InfluxDBClientFactory.Create(url, username, password.ToCharArray());
+
+            //
+            // Get ID of Organization with specified name
+            //
+            var orgId = (await client.GetOrganizationsApi().FindOrganizationsAsync(org: org)).First().Id;
```
Comment (bednar) [inline Examples/RecordRowExample.cs:25]: Please use organisation name instead of ID:

```suggestion
```

File: Examples/RecordRowExample.cs [hunk 8]
```diff
@@ -0,0 +1,67 @@
...
+
+            //
+            // Query data with pivot
+            //
+            var queryApi = client.GetQueryApi();
+            var fluxQuery = $"from(bucket: \"{bucket}\")\n"
+                            + " |> range(start: -1m)"
+                            + " |> filter(fn: (r) => (r[\"_measurement\"] == \"point\"))"
+                            + " |> pivot(rowKey:[\"_time\"], columnKey: [\"_field\"], valueColumn: \"_value\")";
+            var tables = await queryApi.QueryAsync(fluxQuery, orgId);
```
Comment (bednar) [inline Examples/RecordRowExample.cs:42]: Please use organisation name instead of ID:
```suggestion
            var tables = await queryApi.QueryAsync(fluxQuery, org);
```

File: Examples/RecordRowExample.cs [hunk 9]
```diff
@@ -0,0 +1,67 @@
...
+
+namespace Examples
+{
+    public static class RecordRowExample
+    {
+        public static async Task Main()
+        {
+            const string url = "http://localhost:9999/";
+            const string username = "my-user";
+            const string password = "my-password";
```
Comment (bednar) [inline Examples/RecordRowExample.cs:16]: Please use authorisation via token, its preferred way:

```suggestion
            const string token = "my-token";
```

File: Examples/RecordRowExample.cs [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+            const string bucket = "my-bucket";
+            const string org = "my-org";
+
+            using var client = InfluxDBClientFactory.Create(url, username, password.ToCharArray());
```
Comment (bednar) [inline Examples/RecordRowExample.cs:20]: Please use authorisation via token, its preferred way:
```suggestion
            using var client = InfluxDBClientFactory.Create(url, token.ToCharArray());
```

**Review by bednar — CHANGES_REQUESTED** [review]

**Review by bednar — APPROVED** [review]
LGTM 🚀

**Review by bednar — APPROVED** [review]
